Cython/Compiler/ExprNodes.py

```diff
@@ -2160,7 +2155,7 @@ def _analyse_target_declaration(self, env, is_assignment_expression):
             self.entry.known_standard_library_import = ""  # already exists somewhere and so is now ambiguous
         if not self.entry and self.annotation is not None:
             # name : type = ...
-            is_dataclass = 'dataclasses.dataclass' in env.directives
+            is_dataclass = env.is_c_class_scope and env.is_dataclass
```

Comment:
Do we need both conditions? Doesn't `is_dataclass` always imply `is_c_class_scope`? Or are you anticipating some kind of support for Python dataclasses as well?

Reply:
It's more that py class scopes won't have the attribute

Reply:
Ah, ok, that makes sense.

However, we already have a special `is_closure_class_scope`, so maybe we could call this one `is_c_dataclass_scope` and promote both to `Scope` class attributes? Maybe they are just special enough to make them independent of `is_c_class_scope`.